scripts/budo.js

```diff
@@ -12,29 +12,51 @@ function execCmd (cmd) {
   return p;
 }
 
-var app = budo('./index.js:build/aframe.js', {
+var consts = {
+  LIVE: process.env.LIVE,  // To enable: `LIVE=1 npm start`
+  NAME: 'AFRAME',
+  ENTRY: './src/index.js',
+  DIST: 'dist/aframe.js',
+  BUILD: 'build/aframe.js',
+  WATCH: '{elements,examples,lib,src}/**/*',
+  PORT: 9000
+};
+
+var opts = {
   debug: process.env.NODE_ENVIRONMENT !== 'production',
   verbose: true,
-  stream: process.stdout,  // log to stdout
-  port: 9000,
-  browserifyArgs: '-s AFRAME'.split(' ')
-});
-
-app
-.watch('**/*.{css,js,html}')
-.live()
-.on('watch', function (eventType, fn) {
-  if (eventType !== 'change' && eventType !== 'add') { return; }
-
-  if (path.extname(fn) === '.css') {
-    // We want to trigger a reload of the entire document, since
-    // browserify-css injects CSS into the page.
-    app.reload();
-  } else if (path.extname(fn) === '.js') {
-    app.reload(fn);
-  } else if (fn.indexOf('elements/templates') === 0) {
-    app.reload();
+  open: process.env.OPEN,  // To enable: `OPEN=1 npm start`
+  stream: process.stdout,
+  host: process.env.HOST,
+  port: process.env.PORT || consts.PORT,
+  browserifyArgs: ['-s', consts.NAME],
+  middleware: function (req, res, next) {
+    // Route `dist/aframe.js` to `build/aframe.js` so we can
+    // dev against the examples :)
+    var path = urlParse(req.url).pathname;
+    if (path.indexOf('/' + consts.DIST) !== -1) {
+      req.url = req.url.replace('/dist/', '/build/');
+    }
+    // TODO: Consider adding middleware that targets specific directories
```

Comment:
I think this is only possibly by extending `livereload-js`. Calling `budo.reload()` just ends up telling the live reload server to trigger a [full page reload](https://github.com/livereload/livereload-js/blob/e1d943628005ad8d18a50ee1e8c29858ca748d10/src/reloader.coffee#L95), so it ends up affecting all open tabs on the same host + port.

Maybe the `livereload-js` server could have a `reloadHref(regexToMatch)` function which only triggers a page reload when the client's `location.href` matches the regex.